wordpress-mobile/WordPressCom-Stats-iOS
# Posting activity for Stats

This PR adds the posting activity graphs to the Stats project:

**On the insights tab:**
<img width="150" alt="scaled screen shot 2016-04-11 at 8 59 36 am" src="https://cloud.githubusercontent.com/assets/154014/14449380/ecfd54e2-0037-11e6-8caa-8404d4f777db.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 01 32 am" src="https://cloud.githubusercontent.com/assets/154014/14449382/ed14319e-0037-11e6-974e-35f6a58e0c10.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 04 21 am" src="https://cloud.githubusercontent.com/assets/154014/14449381/ed1289ca-0037-11e6-936b-cad4a95dcd7e.png"><img width="225" alt="scaled screen shot 2016-04-11 at 9 00 37 am" src="https://cloud.githubusercontent.com/assets/154014/14449383/ed157496-0037-11e6-89a3-a8814d4b9cb5.png">

Tapping on the cell in the insights table presents a full screen view:

**Full screen mode:**
<img width="150" alt="scaled screen shot 2016-04-11 at 8 59 54 am" src="https://cloud.githubusercontent.com/assets/154014/14449391/0df80bce-0038-11e6-98e2-9a606ead32a8.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 00 13 am" src="https://cloud.githubusercontent.com/assets/154014/14449390/0df6ee24-0038-11e6-800f-e04f3fbc8daa.png"><img width="225" alt="scaled screen shot 2016-04-11 at 9 00 24 am" src="https://cloud.githubusercontent.com/assets/154014/14449393/0df9462e-0038-11e6-844e-46ca264db1f0.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 01 50 am" src="https://cloud.githubusercontent.com/assets/154014/14449392/0df93940-0038-11e6-999f-8a739653f1b9.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 02 28 am" src="https://cloud.githubusercontent.com/assets/154014/14449394/0dfa6e6e-0038-11e6-876e-e963a5e3b7c1.png"><img width="150" alt="scaled screen shot 2016-04-11 at 9 04 08 am" src="https://cloud.githubusercontent.com/assets/154014/14449395/0dfc4df6-0038-11e6-9c2d-93036a68caeb.png">

In the full screen view, the user can tap any individual cell and the date is presented in the header (and the cell is highlighted orange).

**Things to note:**
* The cells are probably too small, but in discussions with @astralbodies, they may be ok for a first version of this feature. Feedback is welcome.
* The iPad and 6s Plus screens should probably have a different layout to allow for more months to be visible. We can tackle this in a subsequent PR.
* Tapping the the actual graph **or** title on the insights table cell launches the full screen mode. I saw this on a couple other cells, not sure if it should be the same here or not.
* I grabbed the cell colors from the web's posting activity graph. For the other parts (e.g. header and footers) I used colors from the stats styleguide already in the iOS project.

**How to test:**
* Try this graph on a wide variety of sites. Make sure the colors and shape of the month is accurate (compare to the web version as needed).
* Test on a wide variety of devices (duh? :grinning: )
* Make sure the insights table scrolls without dropping frames (drawrect should be fast enough now on the contrib graph)
* Be sure to test the "no data" states. Does the posting activity screen/cell behave appropriately?
* What happens if the `/streak` endpoint hangs or returns incomplete data?
* Additional unit tests should be green

Needs Review: @jancavan (design) @astralbodies (code and everything)
/cc @mattmiklic
